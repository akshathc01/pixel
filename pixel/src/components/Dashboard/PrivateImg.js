import React, { useState } from 'react';
import Swal from 'sweetalert2';
import { storage } from '../../firebase/index.js'

const PrivateImg = ({ upload }) => {

    const [ui, setUi] = useState(false)

    var storageRef = storage.ref();


    const handleDelete = (id) => {
        fetch(`http://localhost:3002/deletePrivate/${id}`, {
            method: 'DELETE'
        })
            .then(res => res.json())
            .then(result => {
                Swal.fire({
                    position: 'center',
                    icon: 'success',
                    title: 'Your image has been deleted',
                    showConfirmButton: false,
                    timer: 1500
                })
            })

        var desertRef = storageRef.child(`images/${upload.imageName}`);
        desertRef.delete().then(() => {

            setUi(true)

        }).catch((error) => {
            console.log(error);
        });
    }
    return (
        <>
            {!ui &&
                <div  data-testid="test-PrivateImg" className="card bg-secondary mx-1 my-1" style={{ width: '30%' }}>
                    <img src={upload.url} className="card-img-top h-50" alt={upload.title} />
                    <div className="card-body">
                        <h5 className="card-title text-light text-center">{upload.title}</h5>
                    </div>
                    <div classNameName="card-footer">
                        <button onClick={() => handleDelete(upload._id)} className="btn btn-danger btn-block">Delete</button>
                    </div>
                </div>

            }
        </>
    );
};

export default PrivateImg;