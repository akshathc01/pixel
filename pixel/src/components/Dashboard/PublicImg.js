import React, { useState } from 'react';
import Swal from 'sweetalert2';
import { storage } from '../../firebase/index.js'

const PublicImg = (props) => {
    const [ui, setUi] = useState(false)
    var storageRef = storage.ref();
    const { _id, url, title, imageName } = props.upload
    const handleDelete = (id) => {
        fetch(`http://localhost:3002/deletePublic/${id}`, {
            method: 'DELETE'
        })
            .then(res => res.json())
            .then(result => {
                Swal.fire({
                    position: 'center',
                    icon: 'success',
                    title: 'Your image has been deleted',
                    showConfirmButton: false,
                    timer: 1500
                })
            })
        var desertRef = storageRef.child(`images/${imageName}`);
        desertRef.delete().then(() => {
            setUi(true)

        }).catch((error) => {
            console.log(error);
        });
        console.log(id)
    }


    return (
        <>
        {!ui &&
            <div  data-testid="test-PublicImg" className="card bg-secondary mx-1 my-1" style={{ width: '30%' }}>
                <img src={url} className="card-img-top h-50" alt={title} />
                <div className="card-body">
                    <h5 className="text-light text-center">{title}</h5>
                </div>
                <div classNameName="card-footer">
                    <button onClick={() => handleDelete(_id)} className="btn btn-danger btn-block">Delete</button>
                </div>
            </div>

        }
    </>
    );
};

export default PublicImg;