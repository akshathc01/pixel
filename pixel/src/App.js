import './App.css';
import 'bootstrap/dist/css/bootstrap.min.css';
import React, { createContext, useState } from "react";
import {
  BrowserRouter as Router,
  Switch,
  Route,
} from "react-router-dom";
import Home from './components/Home/Home/Home';
import AddImage from './components/AddImage/AddImage';
import Login from './components/Login/Login';
import Dashboard from './components/Dashboard/Dashboard';
import PrivateRoute from './components/PrivateRoute/PrivateRoute';

export const UserContext = createContext();
function App() {
  const [loggedInUser, setLoggedInUser] = useState({});
  return (
    <div data-testid="test-App">
      <UserContext.Provider value={[loggedInUser, setLoggedInUser]} >
        <Router>
          <Switch>
            <Route exact path="/">
              <Home />
            </Route>
            <PrivateRoute path="/addImage">
              <AddImage />
            </PrivateRoute>
            <PrivateRoute path="/dashBoard">
              <Dashboard />
            </PrivateRoute>
            <Route path="/signedIn">
              <Login />
            </Route>
          </Switch>
        </Router>
      </UserContext.Provider>
    </div>

  );
}

export default App;
