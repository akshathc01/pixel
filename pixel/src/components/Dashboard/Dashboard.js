import React, { useContext, useEffect, useState } from 'react';
import { UserContext } from '../../App';
import Navbar from '../Home/Navbar/Navbar';
import PrivateImg from './PrivateImg';
import PublicImg from './PublicImg';

const Dashboard = () => {
    const [loggedInUser, setLoggedInUser] = useContext(UserContext);
    const [publicImg, setPublicImg] = useState([])
    const [privateImg, setPrivateImg] = useState([])

    useEffect(() => {
        fetch('http://localhost:3002/getCommunity?email=' + loggedInUser.email)
            .then(res => res.json())
            .then(data => setPublicImg(data))
    }, [loggedInUser.email])


    useEffect(() => {
        fetch('http://localhost:3002/getPrivate?email=' + loggedInUser.email)
            .then(res => res.json())
            .then(data => setPrivateImg(data))
    }, [loggedInUser.email])

    return (
        <div data-testid="test-Dashboard">
            <div style={{ background: "#12161f", height: "950px" }} className="text-light">
                <Navbar />
                <h1 className=" pt-5 text-center">Hi {loggedInUser.name} , Here Is your Uploaded Images.</h1>
                <div className="row mt-5">
                    <div className="col-md-6">
                        <h1 className="text-center">Public Images</h1>
                        <div className="row d-flex justify-content-center">
                            {
                                publicImg.map(upload => <PublicImg upload={upload} ></PublicImg>)
                            }
                        </div>

                    </div>
                    <div className="col-md-6">
                        <h1 className="text-center">Private Images</h1>
                        <div className="row d-flex justify-content-center">
                            {
                                privateImg.map(upload => <PrivateImg upload={upload} ></PrivateImg>)
                            }
                        </div>
                    </div>

                </div>
            </div>

        </div>
    );
};

export default Dashboard;