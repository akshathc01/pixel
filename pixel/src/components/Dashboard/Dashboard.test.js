import { render, screen, cleanup } from '@testing-library/react'
import renderer from 'react-test-renderer'
import { UserContext } from '../../App'
import Dashboard from './Dashboard'
import PrivateImg from './PrivateImg'
import PublicImg from './PublicImg'
import { BrowserRouter as Router } from 'react-router-dom';

test('test', () => {
    expect(true).toBe(true)
})

//unit testing and snapshot for dashboard
test('it will render Dashboard component', () => {
    const loggedInUser = {
        email: "pixel@gmail.com",
        name: "Pixel Uploader",
        isSignedIn: true,
        photo: "https://lh3.googleusercontent.com/a-/AOh14GhyyoLqA1sJPhj7Xib2QoaPG1-KARKEMhcQSoYpDA=s96-c"
    }

    render(
        <UserContext.Provider value={[loggedInUser]}>
            <Router>  <Dashboard />  </Router>
        </UserContext.Provider>)
    const dashboardElement = screen.getByTestId('test-Dashboard')
    expect(dashboardElement).toBeInTheDocument()
})

test('matches snapshot for dashboard', () => {
    const loggedInUser = {
        email: "pixel@gmail.com",
        name: "Mohammad Eusha",
        isSignedIn: true,
        photo: "https://lh3.googleusercontent.com/a-/AOh14GhyyoLqA1sJPhj7Xib2QoaPG1-KARKEMhcQSoYpDA=s96-c"
    }
    const tree = renderer.create(
        <UserContext.Provider value={[loggedInUser]}>
            <Router>  <Dashboard />  </Router>
        </UserContext.Provider>
    ).toJSON()
    expect(tree).toMatchSnapshot()
})


//unit testing and snapshot for PrivateImg

test('it will render PrivateImg component', () => {
    const upload = { _id: 1, imageName: '8c5dc61b7c71444ca37bc342020e7891.jpg', url: 'https://firebasestorage.googleapis.com/v0/b/pixel-image-uploader.appspot.com/o/images%2F8c5dc61b7c71444ca37bc342020e7891.jpg?alt=media&token=4fbf17bd-e125-4536-a02b-1f40518dbdcd', email: 'eusha37@gmail.com', title: 'Sleeping Cat' }

    render(<PrivateImg upload={upload} />)
    const dashboardElement = screen.getByTestId('test-PrivateImg')
    expect(dashboardElement).toBeInTheDocument()
})


test('matches snapshot for privateImg', () => {
    const upload = { _id: 1, imageName: '8c5dc61b7c71444ca37bc342020e7891.jpg', url: 'https://firebasestorage.googleapis.com/v0/b/pixel-image-uploader.appspot.com/o/images%2F8c5dc61b7c71444ca37bc342020e7891.jpg?alt=media&token=4fbf17bd-e125-4536-a02b-1f40518dbdcd', email: 'eusha37@gmail.com', title: 'Sleeping Cat' }
    const tree = renderer.create(<PrivateImg upload={upload} />).toJSON()
    expect(tree).toMatchSnapshot()
})


//unit testing and snapshot for PublicImg
test('it will render PublicImg component', () => {
    const upload = { _id: 1, imageName: '8c5dc61b7c71444ca37bc342020e7891.jpg', url: 'https://firebasestorage.googleapis.com/v0/b/pixel-image-uploader.appspot.com/o/images%2F8c5dc61b7c71444ca37bc342020e7891.jpg?alt=media&token=4fbf17bd-e125-4536-a02b-1f40518dbdcd', email: 'eusha37@gmail.com', title: 'Sleeping Cat' }

    render(<PublicImg upload={upload} />)
    const dashboardElement = screen.getByTestId('test-PublicImg')
    expect(dashboardElement).toBeInTheDocument()
})


test('matches snapshot for publicImg', () => {
    const upload = { _id: 1, imageName: '8c5dc61b7c71444ca37bc342020e7891.jpg', url: 'https://firebasestorage.googleapis.com/v0/b/pixel-image-uploader.appspot.com/o/images%2F8c5dc61b7c71444ca37bc342020e7891.jpg?alt=media&token=4fbf17bd-e125-4536-a02b-1f40518dbdcd', email: 'eusha37@gmail.com', title: 'Sleeping Cat' }
    const tree = renderer.create(<PublicImg upload={upload} />).toJSON()
    expect(tree).toMatchSnapshot()
})