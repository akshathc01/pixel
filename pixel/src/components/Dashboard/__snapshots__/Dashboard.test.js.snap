// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot for dashboard 1`] = `
Array [
  "  ",
  <div
    data-testid="test-Dashboard"
  >
    <div
      className="text-light"
      style={
        Object {
          "background": "#12161f",
          "height": "950px",
        }
      }
    >
      <div
        className="full-container"
        data-testid="test-Navbar"
      >
        <nav
          className="navbar navbar-expand-lg navbar-light p-0 bg-dark "
        >
          <div
            className="container-fluid text-light"
          >
            <a
              className="navbar-brand d-flex align-items-center"
              href="/"
              onClick={[Function]}
            >
              <img
                alt="Icon"
                className="logo"
                src="icon.png"
                style={
                  Object {
                    "width": "50px",
                  }
                }
              />
              <h1
                className="pl-3 text-light"
              >
                Pixel
              </h1>
            </a>
            <button
              className="btn btn-danger ml-3"
            >
              <a
                className="text-light"
                href="/addImage"
                onClick={[Function]}
              >
                 Upload Image 
              </a>
            </button>
            <button
              aria-controls="navbarNavAltMarkup"
              aria-expanded="false"
              aria-label="Toggle navigation"
              className="navbar-toggler"
              data-target="#navbarNavAltMarkup"
              data-toggle="collapse"
              type="button"
            >
              <span
                className="navbar-toggler-icon"
              />
            </button>
            <div
              className="collapse navbar-collapse justify-content-end"
              id="navbarNavAltMarkup"
            >
              <div
                className="navbar-nav"
              >
                <a
                  className="nav-link active px-3 text-light h5  my-2"
                  href="/"
                  onClick={[Function]}
                >
                   Home 
                  <span
                    className="sr-only"
                  >
                    (current)
                  </span>
                   
                </a>
                <a
                  className="nav-link px-3 my-2"
                >
                  <a
                    className="text-light h5 "
                    href="/dashboard"
                    onClick={[Function]}
                  >
                    Images
                  </a>
                </a>
                <button
                  className="btn btn-outline-danger mx-3 my-2"
                  type="button"
                >
                   
                  <a
                    className="text-light"
                    href="/signedIn"
                    onClick={[Function]}
                  >
                    Sign In
                  </a>
                    
                </button>
                <button
                  className="btn btn-danger mx-3 my-2"
                >
                  <a
                    className="text-light"
                    href="/signedIn"
                    onClick={[Function]}
                  >
                    Sign Up
                  </a>
                </button>
              </div>
            </div>
          </div>
        </nav>
      </div>
      <h1
        className=" pt-5 text-center"
      >
        Hi 
        Mohammad Eusha
         , Here Is your Uploaded Images.
      </h1>
      <div
        className="row mt-5"
      >
        <div
          className="col-md-6"
        >
          <h1
            className="text-center"
          >
            Public Images
          </h1>
          <div
            className="row d-flex justify-content-center"
          />
        </div>
        <div
          className="col-md-6"
        >
          <h1
            className="text-center"
          >
            Private Images
          </h1>
          <div
            className="row d-flex justify-content-center"
          />
        </div>
      </div>
    </div>
  </div>,
  "  ",
]
`;

exports[`matches snapshot for privateImg 1`] = `
<div
  className="card bg-secondary mx-1 my-1"
  data-testid="test-PrivateImg"
  style={
    Object {
      "width": "30%",
    }
  }
>
  <img
    alt="Sleeping Cat"
    className="card-img-top h-50"
    src="https://firebasestorage.googleapis.com/v0/b/pixel-image-uploader.appspot.com/o/images%2F8c5dc61b7c71444ca37bc342020e7891.jpg?alt=media&token=4fbf17bd-e125-4536-a02b-1f40518dbdcd"
  />
  <div
    className="card-body"
  >
    <h5
      className="card-title text-light text-center"
    >
      Sleeping Cat
    </h5>
  </div>
  <div
    classNameName="card-footer"
  >
    <button
      className="btn btn-danger btn-block"
      onClick={[Function]}
    >
      Delete
    </button>
  </div>
</div>
`;

exports[`matches snapshot for publicImg 1`] = `
<div
  className="card bg-secondary mx-1 my-1"
  data-testid="test-PublicImg"
  style={
    Object {
      "width": "30%",
    }
  }
>
  <img
    alt="Sleeping Cat"
    className="card-img-top h-50"
    src="https://firebasestorage.googleapis.com/v0/b/pixel-image-uploader.appspot.com/o/images%2F8c5dc61b7c71444ca37bc342020e7891.jpg?alt=media&token=4fbf17bd-e125-4536-a02b-1f40518dbdcd"
  />
  <div
    className="card-body"
  >
    <h5
      className="text-light text-center"
    >
      Sleeping Cat
    </h5>
  </div>
  <div
    classNameName="card-footer"
  >
    <button
      className="btn btn-danger btn-block"
      onClick={[Function]}
    >
      Delete
    </button>
  </div>
</div>
`;
